### ISSUE TITLE
训练问题

### ISSUE THREAD

**Issue by plenari** [GuYuc/WS-DAN.PyTorch#10]
训练fgvc-aircraft,40个epoch基本上稳定了，现在raw的正确率82,%，crop正确率74%，drop正确率66%。raw+crop的正确率是78左右。请问这是没有学到注意力吗？

**Comment by GuYuc** [comment]
@plenari 你好。我认为跑40个epoch还是不够的，可以试试用作者发布代码里面的参数配置去训练。我最近在根据作者发布的tensorflow代码试图用pytorch复现文章那四个数据集上的效果，但很可惜全都没有达到。

**Comment by wvinzh** [comment]
我最近也在实现pytorch版本，离作者的还有一定的差距，结果如下：
| Dataset       | ACC    | ACC Refine |
| ------------- | ------ | ----------- |
| CUB-200-2011  | 87.401 | 87.487      |
| Stanford Cars | 92.837 | 93.595      |
| FGVC-Aircraft | 89.319 | 89.769      |

**Comment by GuYuc** [comment]
@wvinzh 你好，非常感谢你分享实验结果。请问可以解释一下你提到的ACC Refine这个指标吗？以及，介绍一下你的实现方法和要点？

**Comment by plenari** [comment]
> @wvinzh 你好，非常感谢你分享实验结果。请问可以解释一下你提到的ACC Refine这个指标吗？以及，介绍一下你的实现方法和要点？



我看ws-dan tensorflow的实现里给每个batch 选择一个attention maps时用到了权重，而不是像torch版本里完全随机选择。并且我感觉从feature map到 feature matrix这样完全线性的变化，完全等同于feature_matrix=conv(1X1)*mean(feature_maps,dim=(1,2,3))*mean(feature_maps,dim=(2,3))。就是把1X1的卷积将两个过程连起来之后，就是用卷积核的第一个通道权重乘以feature_map在2,3维度平均后的值得到第一行的feature_matrix,所以注意力是这个吗？

**Comment by wvinzh** [comment]
@GuYuc 我基本是按照官方的tf版本实现的，前几天发现忽略了center loss 计算的l2-norm, 现在在CUB-200-2011上以及达到89.2的准确率了，Refine就是论文中的Object Localization and Refinement
